### ISSUE TITLE
WAF settings with "/0" in the IP address are refused by data plane

### ISSUE THREAD

**Issue by htech7x** [solo-io/gloo#9204]
### Gloo Edge Product

Enterprise

### Gloo Edge Version

1.16

### Kubernetes Version

1.28

### Describe the bug

Entry "/0" in WAF settings of "ruleSets" is passed through and not caught by control plane,
but is refused by data plane

### Expected Behavior

"/0" will be processed properly

### Steps to reproduce the bug

1. Follow WAF docs, but use "/0" in CIDR for "SecRule REMOTE_ADDR" instead of "/16" https://docs.solo.io/gloo-edge/latest/guides/security/waf/#ip-allowlist
```
vs.yaml
---
apiVersion: gateway.solo.io/v1
kind: VirtualService
metadata:
  name: vs-pets
  namespace: gloo-system
spec:
  virtualHost:
    domains:
      - "*"
    routes:
      - matchers:
        - prefix: /pets
        options:
          prefixRewrite: /api/pets
        routeAction:
          single:
            upstream:
              name: default-petstore-8080
              namespace: gloo-system
    options:
      waf:
        ruleSets:
        - ruleStr: |
            SecRuleEngine On
            SecRule REMOTE_ADDR "!@ipMatch 173.175.0.0/0" "phase:1,deny,status:403,id:1,msg:'block ip'"
```

2.  Apply vs.yaml

```
kubectl apply -f vs.yaml
```
3.  Checks Gloo resources for errors:
```
glooctl check
----------
glooctl binary version (1.15.18) differs from server components (v1.16.0) by at least a minor version.
Consider running:
glooctl upgrade --release=v1.16.0
----------

Checking deployments... OK
Checking pods... OK
Checking upstreams... OK
Checking upstream groups... OK
Checking auth configs... OK
Checking rate limit configs... OK
Checking VirtualHostOptions... OK
Checking RouteOptions... OK
Checking secrets... OK
Checking virtual services... OK
Checking gateways... OK
Checking proxies... 1 Errors!
Gloo has detected that the data plane is out of sync. The following types of resources have not been accepted: [{resource="type.googleapis.com/envoy.config.route.v3.RouteConfiguration"}]. Gloo will not be able to process any other configuration updates until these errors are resolved.
Problem while checking for gloo xds errors
Error: 2 errors occurred:
	* An update to your gateway-proxy deployment was rejected due to schema/validation errors. The envoy_http_rds_update_rejected{envoy_rds_route_config="listener-__-8080-routes",envoy_http_conn_manager_prefix="http"} metric increased.
You may want to try using the `glooctl proxy logs` or `glooctl debug logs` commands.

	* Problem while checking for gloo xds errors

```

4. Check deployment.apps/gateway-proxy logs:
```
kubectl logs -f deployment.apps/gateway-proxy -n gloo-system
...
[2024-02-28 22:59:14.383][1][debug][config] [external/envoy/source/extensions/config_subscription/grpc/grpc_mux_impl.cc:245] Received gRPC message for type.googleapis.com/envoy.config.route.v3.RouteConfiguration at version 8508239728413355047
[2024-02-28 22:59:14.383][1][debug][config] [external/envoy/source/extensions/config_subscription/grpc/grpc_mux_impl.cc:222] Pausing discovery requests for type.googleapis.com/envoy.config.route.v3.RouteConfiguration (previous count 0)
[2024-02-28 22:59:14.384][1][warning][config] [external/envoy/source/extensions/config_subscription/grpc/grpc_subscription_impl.cc:128] gRPC config for type.googleapis.com/envoy.config.route.v3.RouteConfiguration rejected: Rules error. File: <<reference missing or not informed>>. Line: 3. Column: 19.
[2024-02-28 22:59:14.384][1][debug][config] [external/envoy/source/extensions/config_subscription/grpc/grpc_mux_impl.cc:229] Decreasing pause count on discovery requests for type.googleapis.com/envoy.config.route.v3.RouteConfiguration (previous count 1)
[2024-02-28 22:59:14.384][1][debug][config] [external/envoy/source/extensions/config_subscription/grpc/grpc_mux_impl.cc:233] Resuming discovery requests for type.googleapis.com/envoy.config.route.v3.RouteConfiguration
[2024-02-28 22:59:14.485][1][debug][config] [external/envoy/source/extensions/config_subscription/grpc/grpc_mux_impl.cc:245] Received gRPC message for type.googleapis.com/envoy.config.route.v3.RouteConfiguration at version 8508239728413355047
[2024-02-28 22:59:14.485][1][debug][config] [external/envoy/source/extensions/config_subscription/grpc/grpc_mux_impl.cc:222] Pausing discovery requests for type.googleapis.com/envoy.config.route.v3.RouteConfiguration (previous count 0)
[2024-02-28 22:59:14.486][1][warning][config] [external/envoy/source/extensions/config_subscription/grpc/grpc_subscription_impl.cc:128] gRPC config for type.googleapis.com/envoy.config.route.v3.RouteConfiguration rejected: Rules error. File: <<reference missing or not informed>>. Line: 3. Column: 19.
[2024-02-28 22:59:14.486][1][debug][config] [external/envoy/source/extensions/config_subscription/grpc/grpc_mux_impl.cc:229] Decreasing pause count on discovery requests for type.googleapis.com/envoy.config.route.v3.RouteConfiguration (previous count 1)
[2024-02-28 22:59:14.486][1][debug][config] [external/envoy/source/extensions/config_subscription/grpc/grpc_mux_impl.cc:233] Resuming discovery requests for type.googleapis.com/envoy.config.route.v3.RouteConfiguration
...
```

**Comment by DuncanDoyle** [comment]
@htech7x When you state:

```
"/0" will be processed properly
```

... what is the definition of "properly"? 

a. It is an invalid configuration and the Gloo Edge control plane should reject it? I.e. we need to improve validation.
b. It is a valid configuration and the Gloo Edge control plane and Envoy should accept it. I.e. a fix is needed in Envoy/WAF/modsecurity.

I'm not sure what the spec states when it comes to these CIDR notations ....

**Comment by DuncanDoyle** [comment]
The CIDR `/0` notation seems to only be valid in [default routes](https://en.wikipedia.org/wiki/0.0.0.0#:~:text=The%20CIDR%20notation%200.0.0.0/0%20defines%20an%20IP%20block%20containing%20all%20possible%20IP%20addresses.%20It%20is%20commonly%20used%20in%20routing%20to%20depict%20the%20default%20route%20as%20a%20destination%20subnet.%20It%20matches%20all%20addresses%20in%20the%20IPv4%20address%20space%20and%20is%20present%20on%20most%20hosts%2C%20directed%20towards%20a%20local%20router.)

**Comment by davidjumani** [comment]
This fix will be in 1.17.0
